### DISCUSSION TITLE
[@types/react] refs should specify an upper bound instead of lower bound

### DISCUSSION THREAD

**Discussion by d07RiV** [DefinitelyTyped/DefinitelyTyped#65753] [Issues with a @types package]
`div` should be able to accept a `Ref<HTMLElement>`,  but a component that expects a forward ref of type `HTMLElement` should not accept `Ref<HTMLDivElement>`. Currently it's not possible to use a single ref for multiple element types (if we don't know what will be rendered yet).

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
Agreed. Unfortunately, this is how TypeScript works. https://github.com/microsoft/TypeScript/issues/30748 has more context on this limitation.
